packages/color-modes/src/index.tsx

```diff
@@ -242,29 +254,60 @@ export const ColorModeProvider: React.FC = ({ children }) => {
     return res
   }, [colorMode, outerTheme])
 
-  const context = {
-    ...outer,
-    theme,
-    colorMode,
-    setColorMode,
-  }
+  return theme
+}
+
+function GlobalColorStyles({ theme }: { theme: Theme }) {
+  return jsx(Global, {
+    styles: () => {
+      return { html: createColorStyles(theme) }
+    },
+  })
+}
+
+function NestedColorModeProvider({
+  outerCtx,
+  children,
+}: {
+  outerCtx: ThemeUIContextValue
+  children: React.ReactNode
+}) {
+  const newTheme = useThemeWithAppliedColorMode({
+    outerTheme: outerCtx.theme,
+    colorMode: outerCtx.colorMode,
+  })
+
+  return (
+    <ThemeUIInternalBaseThemeProvider
+      context={{ ...outerCtx, theme: newTheme }}>
+      {/* Changed CSS Variables will cascade from the wrapping div */}
+      {jsx('div', {
```

Comment:
We could make this a `section` tag.
Not a big deal if it's a div, but would probably be more syntactically right.

Reply:
I think I'd use a div here, because I truly only need it to apply styles, so want it to be as "semantically transparent" as possible.

I'm a bit worried that nested providers will now affect user's DOM and make layout less obvious. We could add "as" prop, accept styles from props, and expose NestedColorThemeProvider in public API, but I'd rather avoid it until people ask for it a lot.